Prevent libraries that use cmake to find libmongoc from being overlinked.
https://jira.mongodb.org/browse/CDRIVER-3431
--- src/libmongoc/build/cmake/libmongoc-1.0-config.cmake.in.orig	2019-11-06 17:21:28.000000000 -0600
+++ src/libmongoc/build/cmake/libmongoc-1.0-config.cmake.in	2019-11-17 12:10:45.000000000 -0600
@@ -31,23 +31,4 @@
 find_library (MONGOC_LIBRARY mongoc-1.0 PATHS "@PACKAGE_LIBRARY_INSTALL_DIRS@" NO_DEFAULT_PATH)
 set (MONGOC_LIBRARIES ${MONGOC_LIBRARY} ${BSON_LIBRARIES})
 
-# If this file is generated by the Autotools on Mac, SSL_LIBRARIES might be
-# "-framework CoreFoundation -framework Security". Split into a CMake array
-# like "-framework CoreFoundation;-framework Security".
-set (IS_FRAMEWORK_VAR 0)
-foreach (LIB
-   @SASL_LIBRARIES@ @SSL_LIBRARIES@ @SHM_LIBRARIES@ @RESOLV_LIBRARIES@
-   @SNAPPY_LIBRARIES@ @ICU_LIBRARIES@
-)
-   if (LIB STREQUAL "-framework")
-      set (IS_FRAMEWORK_VAR 1)
-      continue ()
-   elseif (IS_FRAMEWORK_VAR)
-      list (APPEND MONGOC_LIBRARIES "-framework ${LIB}")
-      set (IS_FRAMEWORK_VAR 0)
-   else ()
-      list (APPEND MONGOC_LIBRARIES ${LIB})
-   endif ()
-endforeach ()
-
 set (MONGOC_DEFINITIONS ${BSON_DEFINITIONS})
